File: Producto/backend/src/main/java/backend/com/produccion/application/dto/RecepcionOCItemDTO.java
package backend.com.produccion.application.dto;

import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.PositiveOrZero;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;

@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class RecepcionOCItemDTO {

    private Long idRecepcionItem;
    private Long recepcionId;

    @NotNull(message = "El ocItemId es obligatorio")
    private Long ocItemId;

    @NotNull(message = "La cantidadRecibida es obligatoria")
    @PositiveOrZero
    private BigDecimal cantidadRecibida;

    @PositiveOrZero
    private BigDecimal cantidadConforme;

    @PositiveOrZero
    private BigDecimal cantidadRechazada;

    private String motivoRechazo;
}

File: Producto/frontend/src/remote/DTO/RecepcionOCItemDTO.js
/**
 * DTO de un ítem dentro de una recepción de OC.
 * Espejo de RecepcionOCItemDTO.java del backend.
 */
export class RecepcionOCItemDTO {
    constructor(data = {}) {
        this.idRecepcionItem = data.idRecepcionItem ?? null;
        this.recepcionId = data.recepcionId ?? null;
        this.ocItemId = data.ocItemId ?? null;
        this.cantidadRecibida = data.cantidadRecibida ?? 0;
        this.cantidadConforme = data.cantidadConforme ?? 0;
        this.cantidadRechazada = data.cantidadRechazada ?? 0;
        this.motivoRechazo = data.motivoRechazo ?? '';
    }
}
